### PR TITLE
Revivability Update

### PR DESCRIPTION
**Pull request by Ty-Omaha**
**What does this PR do:**
Attempts to add more immersion and RP to the game by increasing negatives for dying.
Reducing "oh no I ded just gotta wait to be cloned/revived/whatever" attitude, hopefully changing to "Hell no, I'm not doing that! I'd like to live thank you very much."

Essentially, making people try to avoid death more, being less like characters in a video game and more like people.

Also removes a bunch of implied `src`'s.

Approved by Necaladun

Should have further testing to make sure I didn't mess anything up, did the best I could without multiple people.
### Cloning

1. Cloning takes 2x as long, upgrade included
2. Pre-scanning removed
3. Autocloning removed

### Strange Reagent (update 1)
Strange Reagent used on species that _can_ be cloned will do nothing (gibbing removed for that)

:cl:
balance: Cloning takes 2x longer
balance: Pre-scanning removed from cloning
balance: Auto-cloning removed
balance: Strange Reagent used on species that can be cloned will do nothing
/:cl:

### CODE REVIEW

**Comment by MrMagolor** [comment]
Maybe people will bring dead people to robotics more now

**Comment by TDSSS** [comment]
While I agree with the rough goal of this (make death more meaningful), I think the methods employed for it are a bit too heavy-handed. Let me go through your changes one by one.

>balance: Cloning takes 2x longer
I'm on board with this one. Cloning is too fast. I could even see it taking even longer.

>balance: Pre-scanning removed from cloning
I think just removing a feature entirely is too much. Why not lock this behind upgrades, requiring a high/top level research components?

>balance: Auto-cloning removed
A bit eh about this, but I can see why it needs to go, tbh.

>balance: Strange Reagent used on species that can be cloned will result in them being gibbed
😕 I really don't see the logic behind this change. SR is a hard to make reagent that requires chemists to jump through a bunch of hoops, if they want to waste it on humans, let them. Not to mention the many, many, many cases of malpractise bay this will cause. 
And what is the lore justification for this? This magical chemical can revive the dead...but not humans, those just explode? 


IMO, if you want to make death matter then make people actually work for cloning. Remove the cloning setup at roundstart and require engineering to build medbay a cloner, maybe lock the cloner board behind some research too so science has to spend a few minutes on RnD.


For these reasons this gets a 👎 from me for now.

File: code/game/machinery/computer/cloning.dm [hunk 1]
```diff
@@ -31,21 +30,21 @@
 	return ..()
 
 /obj/machinery/computer/cloning/process()
-	if(!scanner || !pods.len || !autoprocess || stat & NOPOWER)
+	if(!scanner || !pods.len || stat & NOPOWER)
 		return
 
-	if(scanner.occupant && can_autoprocess())
+	if(scanner.occupant)
 		scan_mob(scanner.occupant)
```
Comment (Ty-Omaha) [inline code/game/machinery/computer/cloning.dm]: I think I actually have to remove this entirely

**Comment by MrMagolor** [comment]
FYI, Ty confirmed in the discord that using data disks to store cloning records will still be doable.

**Comment by datlo** [comment]
Can't support the current SR change, it's clunky and doesn't make much sense.

If you want to remove SR but keep it for vox and slimes, then add new methods to revive vox and slimes and rip SR out entirely.

**Comment by Ty-Omaha** [comment]
>FYI, Ty confirmed in the discord that using data disks to store cloning records will still be doable.



I dont think its actually possible to load a genetics disk into a cloner so I will have to test later

**Comment by MINIMAN10000** [comment]
I always like to think about the broad scope of what I think and compare it against PRs and throw in my two cents.
So as a broad scope, I believe people should always have a way back to their profession. They choose that profession because they had something they wanted to do in that round so let them do it.
Prescanning made this easy, any time you died you were always loaded into the scanner and thus always had a way to jump right back into your body and continue where you left off.
So as we can see there is a conflict here between my broad scope and this PR.
Removing prescanning is the only broad scope issue I hold with this PR and I'll leave discussion of changes on other changes this PR makes to everyone else.

Now let me put in my two cents on suggestions within this PR related to this.
TDSSS suggests that prescanning require a high tier research. This fulfills me request of "having a way back" it just requires research.
Being able to load a data disk into the cloner and clone someone through that also fulfills this. Currently there is no way to clone from a data disk.

**Comment by Mitchs98** [comment]
I don't understand the logic behind making SR gib cloneable races, especially since this would include plasamamen. I see no reason to give that big of a fuck you to them, at all. The SR changes just seem tacked on for very little reason. 

That said the suggestion that was made to make it harder to revive non-cloneable races just isn't viable. Nor is removing SR and getting different methods. Medbay is already bad enough as it is, there is NO reason to make it harder for people to get back into the game solely because of medbay's incompetency.

That said.

Longer cloning times are doable, actually a good thing when medbay is hectic and you don't have time to monitor cloning as much.

Prescanning and autocloning should be locked behind tech levels, not outright removed. That's just all kinds of wrong. Autocloning is already infact restricted by tech. If its that much of a problem raise the tech levels, not remove a feature because that's the easiest solution to do.

Overall. Don't support this for obvious reasons including the fact changing SR could overall provoke more changes to make revival of uncloneable races harder for no good reason when they already rely on medbay to know what they're doing.

**Comment by Kyep** [comment]
I wonder what would happen if, instead of gibbing corpses with excessive damage, SR made them revive as a hostile zombie mob instead. Not sure if its doable, let alone desirable, but it would introduce negative consequences for the misapplication of SR. Negative consequences for the docs doing it, I mean, not just the corpse of the person they try it on.

**Comment by Bookdrizzt** [comment]
I’d support Kyep’s idea so long as the zombie could be brought back through proper methods later.

**Comment by Allfd** [comment]
Rule of zombies, a zombie is a zombie forever.  Your family member is dead, the thing the body is now will eat your face first chance it gets.

**Comment by Shadeykins** [comment]
> That said the suggestion that was made to make it harder to revive non-cloneable races just isn't viable. Nor is removing SR and getting different methods. Medbay is already bad enough as it is, there is NO reason to make it harder for people to get back into the game solely because of medbay's incompetency.

It's perfectly viable, it was in fact the case on the server for the majority of it's lifespan. I'm really uncertain why you believe this notion, which was our standard prior to Goonchem, is untenable.

It works just fine. There are several years of data to reinforce that.

**Comment by DesolateG** [comment]
Look, you want to mess with cloning? I have a much better solution. Make scans stay permanently, and each time you're cloned it gets a larger genetic damage multiplier. Then, remove genetic damage healing from cryoxodone. Now medical will need rezadone, which is dangerous like SR. And to make rezadone you need carpotoxin, which is only availible from space carps, or koibeans from botany. Now we have even more departments cooperation.

Strange Reagents shouldn't even  be touched; their previous nerf already made it much more of a process to bring someone back with it.

Removing Auto cloning isn't just a tweak it's a feature removal, and it won't remove incentive not to die, only incentive for science to listen to medbay crying for upgrades.

**Comment by Mitchs98** [comment]
I absolutely disagree with Desolates proposed idea entirely. That's outright worse than the current PR and makes death more than just 'more impactful'. That makes you have to rely on incompetent people more than you already have to do. It's also a full on nerf to people that have to use SR to be revived. You already have to rely on medbay to even make SR. You don't nor should you need to rely on them to make an obscure chem that needs other obscure chems to even properly revive them.

The SR change is tacked on in this PR regardless. It's added without taking into account you have to spend more time fixing people when you use it than you do cloning them. I don't even understand why it was added in the same PR.

Also Shady. That may of been true in the past, but I see no reason to go back to it. Especially if its something overly complicated and convoluted just to 'make death more impactful'.

**Comment by DesolateG** [comment]
The only reason rezadone is "obscure" is because cryoxodone is the end all be all easy to make healing chem. Shock em to life, toss em in the cryo tube. Hell it even  will keep someone having heart attacks alive so long as they're in there.

It's not hard to make, and koibeans are easy to grow. My suggestion doesn't make it "hellish" as you over dramatically imply.

**Comment by Agameofscones** [comment]
>Be security main
>Find changeling that is human/plasmameme/skrell/clone-able
~~>Keep them batoned/tased/memechemed down
>Get your buddy to feed the SR~~
>Use a syringe gun loaded with SR
>They gib without recourse since the damage threshold is gone
>Get all the medals
>kthnxbye

Jokes aside, the implications in regards to changelings warrants discussion in its own right. This changes the play around an antag that normally only sees permanent removal via decapitation or incineration, and whether this has been considered as a new avenue for countering them (Which I do personally support, since I find changelings to be very difficult to defeat, and not every security officer has chapel access and or carries a circular saw in their back pocket). Perhaps the SR change should be considered for another PR.

The other changes are, interesting to say the least, I do echo the earlier sentiments that pre-scanning should be upgrade locked, and not removed entirely. _Maybe_ same with auto-cloning as well, but that sort of implies a research system where ProRDMan(tm) doesn't get 7's on every tech in 30 minutes.

**Comment by farie82** [comment]
I myself am for making death more impactful. I don't mind the cloning things but the SR is just a pain in the ass. You have to reward doctors that use medicine to revive somebody instead of punish them. Healing somebody up till they are SRable takes a doctor. This takes a lot of medbays time and resources.
I'd myself prefer if the SR punish is removed from this PR due to that.

**Comment by dovydas12345** [comment]
I am fine with having the cloning changes made, but i dont think this specific SR change is right. I feel like it should be changed differently.

**Comment by Ralta** [comment]
All for making death more impactful, but not sure that this is the right way to go about it, nor that it will make people reconsider dangerous actions so as to stay alive.

In fact, it might reinforce powergaming behaviour from people, encouraging them to get saline and other healing chems to keep themselves alive, along with whatever else they feel they can do to prevent an untimely demise.

I saw this has gone in for testing, so that will be helpful because it's hard to predict what the actual outcome will be. But I think it's important that we look at not only what it changes in medical, but also the intended effect it has on peoples actions in terms of avoiding death. Whether that means avoiding dangerous situations, or ensuring they are much more prepared for those situations.

Any chance we could get a post-test survey for any test runs? I know this isn't _the biggest PR change of all time, EVER_, but it IS a bit controversial already, and I would be curious if the playerbase feels they've changed their approach for the tests. In that vein, it might be worth making a Discord announcement about the tests so it's clear when they are/are not happening.

**Comment by Ty-Omaha** [comment]
> In that vein, it might be worth making a Discord announcement about the tests so it's clear when they are/are not happening.

There is a development news channel that does exactly that

**Comment by Ty-Omaha** [comment]
### SR Update
SR no longer gibs and just doesn't work on clonable races

**Comment by Ralta** [comment]
> > In that vein, it might be worth making a Discord announcement about the tests so it's clear when they are/are not happening.
> 
> There is a development news channel that does exactly that

I am aware, but it has not pinged @everyone, so I imagine most will be unaware of this test. Furthermore, I am suggesting the idea of a post-test survey, so asking people to have that in the back of their minds during the test is probably a good idea.

**Comment by farie82** [comment]
I'm still of the opinion that SR should still be able to revive somebody. Good that the gibbing part is gone.
This will make it harder for cloneable species to get back. Indirectly buffing non cloneable species. I'm not for buffing nor debuffing either but this is something to keep in mind.

**Comment by TDSSS** [comment]
So what is the justification behind making SR only work on unclonable races? Changing it to not gib is a move in the right direction but it still doesn't make sense that the magical revival chem somehow works on dogs and vox and slimes but not monkeys and humans or whatever.

**Comment by Mitchs98** [comment]
Agreeing with Farie once again. Its good that it doesn't gib them. But it should still revive them. It makes absolutely no sense that it doesn't. It still takes a lot of time and effort to revive someone with SR, especially if they've been dead a while. Not to mention a competent enough doctor to know what they're doing. It also requires the 'interdepartmental co-operation' everyone likes I.E getting monkies from genetics, science, botany, or cargo for organs.

SR isn't just a 'Slap it in and you're good bruh' revival chem. It takes, sometimes a LOT, of time to get the people it revives up and going again. I see no reason to punish a competent medical staff for knowing how to do their job. Not to mention its entirely illogical it doesn't work.

**Comment by Allfd** [comment]
Current discussion on the forums seem to be in the direction of just removing SR

**Comment by Aslan4468** [comment]
I agree with Moi on all they said about strange reagent. All the cloning changes seem perfectly fine to me, but in my opinion SR is already balanced as is, as it often requires more time than cloning to fix. It requires chemistry (Mitocholide) or genetics (Spare organs/limbs) to get the parts required to fix all the septic issues with the person, as well as a competent surgeon that understands how to use all these things properly. I see no reason to make strange reagent only work on the unclonable species.

**Comment by Spacemanspark** [comment]
Honestly?
Remove SR completely.

Let these new kiddos know about the days when such magic did not exist within our realm.

**Comment by Shadeykins** [comment]
> Also Shady. That may of been true in the past, but I see no reason to go back to it. Especially if its something overly complicated and convoluted just to 'make death more impactful'.

There's no convolution in this approach sans the thing with SR. Lacking SR would be less convoluted, there would be less complexity when it comes to the revival process and less for new doctors to learn/handle.

Death has been completely trivialized and the result has been a pretty severe uptick in chaotic behaviour. Killing someone doesn't terribly manner, death is not something to be avoided, and people feel entitled to being revived because it's "easy." I personally view this mode of behaviour as being due to a combination of nerfing atmos (LINDA, Holobarriers) and buffing medical (SR, Defibs, Mitocholide, etc.).

I would go as far to hypothesize that people would be less likely to chase antags down maintenance if they knew full well that getting a revolver to the face may mean actually sitting out for more than three minutes. Paranoia was a huge part of our server atmosphere awhile back and this lead to a variety of interesting interactions, this has mostly been removed due to an abundance of anti-death/revival options.

**Comment by EvadableMoxie** [comment]
If SR is removed but cloning is left in, that change your looking for is really only going to apply to Vox and Slime people, and only Vox and Slime people who don't just choose to switch to a different race. Everyone else is still going to be clonable, the only difference is it'll take a bit longer.  Most people don't pre-scan anyway, and auto-process without pre-scanning is meaningless since if someone is there to put your body in the tube they'll be there to hit the start button.  So really we're just doubling the clone time which is already incredibly short when fully upgraded.

I think cloning and SR should be a package deal.  Either both go or both are left in.  Getting rid of one without the other is trying to apply a change to the entire player base by targeting two specific races that make up a small fraction of it.

**Comment by Pennwick** [comment]
> 
> 
> Honestly?
> Remove SR completely.
> 
> Let these new kiddos know about the days when such magic did not exist within our realm.

BRING
BACK
TREKCHEM

Really though I wouldn't be too sad seeing it go or seeing it gated behind something more than just donk pocket dialysis and a bottle of wine. If it was something like diamond dust and powdered slime cores it might be seen as more of a rarity.

As for the other cloning changes I'd be sad to see prescanning go but really its a bit of an RP nightmare as is. You logically shouldn't remember anything from before your scanning but thats just like trying to police Clone Memory Disorder. I'd like to see Autoprocessing stay though but you can't feasably do one without the other. I just want geneticists to load 3 corpses into the cloner and not have to babysit the thing to make sure they're being efficent. Especially if clone times are doubled, any time not cloning it serious time lost. Perhaps just a buffer of one is enough to solve that issue. (One in the tank, one in the scanner, kind of deal. Clone pops out, next one starts.)

**Comment by Allfd** [comment]
> If SR is removed but cloning is left in, that change your looking for is really only going to apply to Vox and Slime people, and only Vox and Slime people who don't just choose to switch to a different race

When I started being a Vox regular, we only started with enough nitrogen to go half the shift, death was death (no zaping or SRing back to life) and the drugs for oxyloss, killed you.

So you could be dying, and then the medical ERT would inject you full of toxins trying to be helpful.

We also had ZAS, so being spaceproof actually meant something.  Also for a while.... we had Vox leap...

A number of planned and unanticipated buffs and nerfs occurred that resulted in us being here now.

Most of Vox culture developed under those conditions.

Granted viruses also used to be tied to species, with Vox being immune from almost all but the most dangerous ones.

**Comment by Jazz23** [comment]
> SR is a hard to make reagent that requires chemists to jump through a bunch of hoops, if they want to waste it on humans, let them.

Go CMO, go to bar and get wine, 90 pills 2 minutes into round start. Every time I go CMO SR is the first chem in the fridge.

**Comment by KasparoVy** [comment]
Is it possible to nerf resuscitation so hard we never lived to begin with?

Technically yes I guess if it keeps players out or makes players leave.

Unfortunately even after playing noclone races for as long as I have I can't gauge for certain how I'd feel about this long term.

All I can tell from the in game testing is that more bodies get left unattended and less end up being clonable due to the player dipping out, myself included. The satisfaction of being a hardcore badasd who survived when everyone else got miserably punished wore off right away

**Comment by Dinarzad** [comment]
If you're trying to make death impactful, this won;t do it.
What this WILL do is cause an increase in powergaming type behaviour as people look for every possible advantage to milk so that they don't get mega-death'd.

You want death to matter? You want roleplay about death and for it to be this big deal?
Then you need to actually enforce Clone Memory Disorder, or rather, bring back the rule when it WAS enforced a bit more. As a self-proclaimed Medium Roleplay server there should probably be some enforced Roleplay rules, rather then crossing fingers an hope people do it.

TL;DR:  This won't solve your problem, this will create new problems while making existing problems even worse.  Increased clone time isn't terrible, all the other stuff kind of is.

**Comment by Citinited** [comment]
> actually enforce Clone Memory Disorder, or rather, bring back the rule when it WAS enforced a bit more.

CMD was never enforced on para (or at least, if it was then it was at the absolute start of the server)

If your goal is to make death matter, maybe increase the resources required to revive someone from the dead or make it harder to do (1u of SR is laughably easy to make, cloning takes maybe 1 minute to complete and is available roundstart, using a defib uses a bit of power and requires you to heal them beforehand and is also available roundstart). Or apply permanent debuffs on the revived mob that they can't get rid of, or something like that. IDK though this isn't something I care to look deeply into :smile:

**Comment by Shadeykins** [comment]
> You want death to matter? You want more roleplay about death and for it to be this big deal?
> Then you need to something most will consider a heretical sin, and actually enforce Clone Memory Disorder, or rather, bring back the rule when it WAS enforced a bit more. As a self-proclaimed Medium Roleplay server there should probably be some enforced roleplay rules, rather then crossing fingers an hope people do it. Sometimes you have to enforce a standard not just hope one comes up.

CMD was never enforced, there was an article on the wiki suggesting it was mandatory. That was about it. The only very specific RP-oriented rule we used to have was restrictions on departmental knowledge.

**Comment by Jazz23** [comment]
M8be give them a permanent random disability?

**Comment by Citinited** [comment]
> M8be give them a permanent random disability?

Yeah, that's what I had in mind - that or code in some new disabilities that are unique to revived mobs. Examples off the top of my head: Muscle stiffening (permanent slowdown), Retinal damage (permanent short-sightedness & colourblind), Kidney damage (toxins process much more slowly than normal), Nerve damage (Occasionally collapse & jittery), Shortness of breath (can't talk louder than a whisper, recover oxygen damage much more slowly than normal).

**Comment by Spacemanspark** [comment]
CMD was never enforced, merely encouraged. Entirely optional, and it eventually stopped being a thing as _players_ stopped doing it.

**Comment by Mitchs98** [comment]
I quite honestly don't agree. Vox and Slime already have it bad enough trying to get revived. There shouldn't be 'further consequences'. That's an absolutely dumb line of thinking. Doesn't really matter if you've been playing a lot of slime people. The majority of slime players don't have git nor the forums to comment, staff and head staff shouldn't get more sway on something like this just because of 1. What you just said. And 2. 'It used to be this way.' It doesn't matter that Slim not Vox were considered in the original PR. I can't understand peoples' thinking on this. Just because something used to be that way doesn't mean it's desirable now. That's an extremely poor line of thinking.

Progress is meant to move forward. Not regress a feature just because you can't think of a solution to a problem.

All you're going to do with removing SR is turn away a portion of those loyal players.

**Comment by Spacemanspark** [comment]
We aren't forcing you to play these races. If you don't like their features or benefits/ draws, you're welcome to try another race on the server.

'Progress' isn't always a straight arrow shot into a bullseye. No, rather, it will be often convoluted and full of backtracking. Sometimes decisions we make aren't the best. Maybe I'm wrong, maybe I'm right.

Won't know unless we try, right?

We _have_ had this feature before... and it was accepted with little issue then. I don't see the issue with it returning.

**Comment by Shadeykins** [comment]
I concur with keeping autocloning in.

Just remove pre-scanning from anything other than disks (which requires you stepping into a machine you can be locked into and murdered with).

**Comment by TullyBurnalot** [comment]
WARNING: ANECDOTAL EXPERIENCE AHEAD

After playing a few rounds in Medical as a Surgeon, both in medium-pop (~50-60 players) and highpop (~100 players), I'm not convinced this... actually does anything? Don't get me wrong, it does exactly what it sets out to do: cloning takes longer and the SR changes force you to queue up folks for cloning, which can certainly become a strain during rounds with high death counts.

That said, it doesn't really... do much, in terms of changing the impact and mechanic of death. Fresh deaths can still just as easily defibbed. If you're a member of a species that can be SR'd, the process remains unaltered; and if you're not, you're not at risk of not getting revived. You just take twice as long to pop out of the magical respawn machine. At the end of the day, while it does exactly what it sets out to do, it does exactly that and just that; you can respawn just as easily provided Medbay isn't crewed by people who skipped every class in their medical degree, and the only real appreciable difference is that, if there isn't any SR around, you just take twice as long to grow from a vat.

Ultimately, I can understand where this game from, but I severely doubt this will do anything but impose artificial restrictions that ultimately don't really lead anywhere. I'd much prefer to see a revamping of the way we handle reviving as an overarching concept rather than this.

**Comment by Raze500** [comment]
**Remove autocloning**

Go to deathclaw instahead if you want that, this is so dumb...

**Comment by Allfd** [comment]
Everyone is aware that this is a larger discussion that is occurring on the forums right?  That it has the backing of a semi-large portion of the staff?

Comments like the last one do not help.

**Comment by Landerlow** [comment]
This update is a definite yes.

**Comment by Ty-Omaha** [comment]
The test of this is completed and better solutions exist such as permanent disabilities and more depth to the cloner instead of number changes
